lisp/ess-r-package.el

```diff
@@ -363,6 +364,20 @@ With prefix argument ARG, run tests on current file only."
 })
 ")
 
+(defun ess-project-save-buffers ()
+  "Offer to save modified files in the current project.
+Respects `ess-save-silently', which see."
+  (let ((cur-proj (project-roots (project-current))))
+    (dolist (buf (buffer-list))
+      (when-let ((file (buffer-file-name buf))
+                 (dir (buffer-local-value 'default-directory buf)))
+        (when (equal cur-proj (ignore-errors
+                                ;; If `project-current' returns nil,
+                                ;; `project-roots' errors.
+                                (project-roots
+                                 (project-current nil dir))))
```

Comment:
This is not quite what I meant. This logic is likely to be extremely slow as project.el doesn't cache anything currently. On remotes it will effectively stall. Worse it might blow on buffers with expired tramp sessions forcing tramp to reconnect.

 We already do cache R projects on file opening and I believe we should reuse it here. That part is not generic, but we could make it with the rename of the `ess-r-package-info-cache` I believe.

Reply:
OK, pushed a small commit that should do the trick I think.